app/main.py

```diff
@@ -16,4 +16,30 @@
 }
 collection_of_coins = {1, 2, 25}
 
-# write your code here
+def sorted_variables_func(*args):
+    types = {"mutable" : ["list", "dict", "set"],
+             "immutable" : ["int", "float", "str", "bool", "tuple"]}
+
+    variables_dict = {key : value for key, value in globals().items()
+                      if not key.startswith("__")}
```

Comment:
The use of `globals()` is correct for accessing global variables, but be cautious as it includes all global variables, which might not be intended. Consider filtering the variables more specifically if needed.